packages/http/server/test/response_builder_spec.mare

```diff
@@ -0,0 +1,66 @@
+:import "spec"
+
+:class ResponseBuilderSpec
+  :is Spec
+  :const describes: "ResponseBuilder"
+
+  :it "builds 1xx responses"
+    builder = ResponseBuilder.new(Bytes.new)
+    builder.status_continue
+
+    // If we wanted to use `@assert = builder.finish == b"..."
+    // we would nead to teach ByteStreamWritable how to use the `==` function:
+    //
+    // :fun "=="(other Bytes'box) Bool: @ == other
```

Comment:
I didn't do this in this PR because I'm don't know if this is something we want in the trait, in part because I don't know enough about traits.

Reply:
Yeah, I don't think we want to add that to the trait - the trait is meant to be generic for anything that can receive Bytes, so adding `==` is not in line with that thinking.

However, in this case you can just hold on to the original `Bytes` object in its original type, unobscured by the trait, to make it easier on yourself.

Something like this:

```
bytes = Bytes.new
builder = ResponseBuilder.new(bytes)
// ...
@assert = bytes == b"HTTP/1.1 100 Continue\r\n\r\n"
```